api/bookmarks/popup.js

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) 2012 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Search the bookmarks when entering the search keyword.
+$('#search').change(function () {
+  $('#bookmarks').empty();
+  dumpBookmarks($('#search').val());
+});
+
+// Traverse the bookmark tree, and print the folder and nodes.
+function dumpBookmarks(query) {
+  var bookmarkTreeNodes = chrome.bookmarks.getTree(function (bookmarkTreeNodes) {
+    $('#bookmarks').append(dumpTreeNodes(bookmarkTreeNodes, query));
+  });
+}
+
+function dumpTreeNodes(bookmarkNodes, query) {
+  var list = $('<ul>');
+  for (var i = 0; i < bookmarkNodes.length; i++) {
+    list.append(dumpNode(bookmarkNodes[i], query));
+  }
+
+  return list;
+}
+
+function dumpNode(bookmarkNode, query) {
+  if (bookmarkNode.title) {
+    if (query && !bookmarkNode.children) {
+      if (String(bookmarkNode.title).indexOf(query) == -1) {
```

Comment:
in testing this, I found it being case-sensitive a bit odd. Can you lower-case both before you check?

Reply:
Done.